pkg/cmd/experimental/syncgroups/config/types.go

```diff
@@ -0,0 +1,148 @@
+package config
+
+import (
+	ldaputil "github.com/openshift/origin/pkg/auth/authenticator/password/ldappassword/util"
+	"github.com/openshift/origin/pkg/client"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+// LDAPSyncCommonConfig contains all of the config that is common to all LDAP schemas
+type LDAPSyncCommonConfig struct {
```

Comment:
> Please explain why.

This abstraction is in the wrong place.  You've created a single uber-object to hold sufficient information to build every piece in this entire flow, but if the constructing code has enough information to build this object and enough insight to organize it all properly, it had enough information organized well enough to construct the sync job itself.  That makes this object an extraneous abstraction that isn't simplifying your code and introduces unusual dependencies and information leaks.